### ISSUE TITLE
ETH holdback warning OR ETH-withdrawal-purchase

### ISSUE THREAD

**Issue by pierrrek** [Loopring/dexwebapp#132]
Hi

In attempting to withdraw USDT I realized I had insufficient ETH. Urgh. This got me thinking.

1. ETH holdback warning
You could warn someone, when they are about to trade out of their last bit of ETH that they will have insufficient ETH left for any withdrawal. I seem to think I saw some message like that when I traded but it's not ideal.

2. ETH-withdrawal-purchase
My preferred option is a cleaner one because it will cover us whenever we have more than one coin we may want to withdraw. A button at the top of the Account Page that becomes available **ONLY** WHEN YOU HAVE **LESS THAN 0.002 ETH** available in your ETH wallet. That button must only appear to allow you to cover **THE BALANCE** of the required withdrawal cost.

In my example below, having 0.0001 ETH available... The button would say "**Purchase 0.0019 Ether to cover withdrawal fee**".
![image](https://user-images.githubusercontent.com/64374649/80403999-d103b180-88c0-11ea-9218-b4e1ddd00e4b.png)

The reason for the hard-line Purchase calculated-limit is because some users may use this opportunity to purchase less than the minimum 0.005 ETH.

On the other hand, you don't want users to have to go back to the trading page and force them to buy 0.005 ETH just so they can cover 0.002 ETH.

How you allow the transaction is up to you. How to decide which pairing to trade on needs further discussion. This brings me to my final option..

3. **UNLESS** YOU HAVE A BUTTON **NEXT TO EACH PAIRING WITH SUFFICIENT FUNDS**. That seems even cleaner. Having said that, once the purchase is done through the chosen PAIR, all buttons next to all Assets disappear because the ETH-withdrawal-purchase is no longer needed.

I hope this is clear.

Thank you,
Pierre

**Comment by pierrrek** [comment]
My understanding of a withdrawal is incorrect, but this got me thinking. You charge 0.002 as withdrawal cost, but the cost of a recent transaction for me was 0.0048 (in addition to the 0.002). That defeats the purpose. Surely. We are paying the network twice the amount of money than you receive. See transaction below. https://etherscan.io/tx/0x1b3d0ef2a34045db8e460dcdfde808504df8bdbae4e95d74453ca4ea9d02b9a7

I don't know if this is a DEX requirement, but surely you could simply get an approval from my wallet - as you do for login - then take the 0.002 from the loopring balance.

I would rather you take 0.005 than 0.002 and only pay the network when doing the actual withdrawal.

**Comment by xiaowheat** [comment]
Please ask this question in https://discord.com/invite/KkYccYp
Thanks
